### ISSUE TITLE
Confusion of algorithm numbers between DAU, DHU, and N3U

### ISSUE THREAD

**Issue by divergentdave** [hickory-dns/hickory-dns#2639]
Currently, the DAU, DHU, and N3U EDNS options are all internally represented with the `SupportedAlgorithms` struct, which is a customized bitfield for DNSSEC signing algorithms. Relatedly, `Edns::set_default_algorithms()` inserts the DAU and DHU options with the same set of algorithms in each. However, algorithm codes have different semantics between the three options, as they are drawn from different IANA registries.

The internal representations of DHU and N3U options should be changed to distinct bitfield types, or lists of `DigestType` and `Nsec3HashAlgorithm`. `Edns::set_default_algorithms()` should report 1, 2, and 4 in the DHU option, for SHA-1, SHA-256, and SHA-384. We could also add the N3U option, with just 1 for SHA-1.

**Comment by bluejekyll** [comment]
I think you cleaned this up recently, right?

**Comment by djc** [comment]
I guess you're referring to #2753?

**Comment by divergentdave** [comment]
Yes, #2753 removed the DHU and N3U options.

**Comment by djc** [comment]
@divergentdave so can this be closed?

**Comment by divergentdave** [comment]
Yes, we can add these options back later if there's interest, and track that in a separate issue.
